src/robusta/core/sinks/robusta/rrm/prometheus_alert_resource_manager.py

```diff
@@ -0,0 +1,209 @@
+                "name": PrometheusAlertResourceManager.__crd_name(slot),
+                "labels": {
+                    "release": "robusta",
+                    "role": "alert-rules",
+                    "release.app": "robusta-resource-management"
+                },
+            },
+
+            "spec": {
+                "groups": [
+                    {
+                        "name": f"rule-{slot}",
+                        "rules": rules
+                    }
+                ]
+            }
+        }
+
+    def create_resource(self, resource: AccountResource) -> Optional[ResourceEntry]:
+        slot = self.__find_available_crd_slot()
+        self.__append_rule(slot, resource)
+        self._timestamp = resource.updated_at
+        return PrometheusAlertResourceEntry(resource=resource, slot=slot)
+
+    def update_resource(self, resource: AccountResource, old_entry: PrometheusAlertResourceEntry) -> Optional[
+        ResourceEntry]:
+        slot = old_entry.slot
+        self.__modify_rule(slot, resource)
+        self._timestamp = resource.updated_at
+        return old_entry
+
+    def delete_resource(self, resource: AccountResource, old_entry: PrometheusAlertResourceEntry) -> bool:
+        slot = old_entry.slot
+        self.__delete_rule(slot, resource)
+        self._timestamp = resource.updated_at
+
+        return True
+
+    # initialize the resources
+    def init_resources(self):
+        try:
+            # fetch the available crd files and then delete them in the first run
+            crd_obj = self.__k8_api.list_namespaced_custom_object(
+                **CRD_PARAMS,
+                label_selector="release.app=robusta-resource-management",
+            )
+
+            items = crd_obj["items"]
+            for obj in items:
+                name = obj["metadata"]["name"]
+                self.__k8_api.delete_namespaced_custom_object(
+                    **CRD_PARAMS,
+                    name=name,
+                    grace_period_seconds=60
+                )
+        except Exception as e:
+            logging.error(f"An error occured while initializing PrometheusRules CRD resources: {e}", exc_info=True)
+
+    # fetch the first available slot in the crd files
+    def __find_available_crd_slot(self) -> int:
+        slot = index_of(self.__cdr_slots_len, lambda l: l < MAX_ALLOWED_RULES_PER_CRD_ALERT)
+        if slot is not None:
+            return slot
+
+        self.__cdr_slots_len.append(0)
+        return len(self.__cdr_slots_len) - 1
+
+    # append rules to the crd files
+    def __append_rule(self, slot: int, resource: AccountResource):
+        name = PrometheusAlertResourceManager.__crd_name(slot)
+
+        if "rule" not in resource.resource_state:
+            resource.resource_state["rule"] = {}
+
+        rule = resource.resource_state.get('rule')
+        if "labels" not in rule:
+            rule["labels"] = {}
+        rule["labels"]["entity_id"] = resource.entity_id
```

Comment:
try to make this stuff shorter (here and in further places). You just want to assign the `entity_id` label.  Examples:
1. `rule = {**rule, {"labels":{"entityId": entity_id}}}`
2. `rule.labels = {**rule.labels, {"entityId": entity_id} }'`

Reply:
done